`ruff check --statistics` shows unsafe fixes as fixable

### Summary

Take the following simple example:

```py
def mvce(keys, values):
    return {key: value for key, value in zip(keys, values)}
```

and run: `ruff check --select=C416 mvce.py`:

```
mvce.py:2:12: C416 Unnecessary dict comprehension (rewrite using `dict()`)
  |
1 | def mvce(keys, values):
2 |     return {key: value for key, value in zip(keys, values)}
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ C416
  |
  = help: Rewrite using `dict()`

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

but running `ruff check --select=C416 --statistics mvce.py`:

```
1       C416    [*] unnecessary-comprehension
[*] fixable with `ruff check --fix`
```

Is this intentional? There _is_ a fix available, but it's unsafe, so at the very least the summary line for `--statistics` is inaccurate in that it also requires `--unsafe-fixes`.

I have a simple fix to only count unsafe fixes if `--unsafe-fixes` is set, and can make a PR if this isn't intentional behaviour.

In either case, should the summary be updated to mention unsafe fixes?

### Version

ruff 0.11.0+2 (631141237 2025-03-14)

--- a/crates/ruff/src/printer.rs
+++ b/crates/ruff/src/printer.rs
@@ -353,7 +353,11 @@ impl Printer {
                 code: message.rule().map(std::convert::Into::into),
                 name: message.kind().into(),
                 count,
-                fixable: message.fixable(),
+                fixable: if let Some(fix) = message.fix() {
+                    fix.applies(self.unsafe_fixes.required_applicability())
+                } else {
+                    false
+                },
             })
             .sorted_by_key(|statistic| Reverse(statistic.count))
             .collect();
@@ -412,9 +416,7 @@ impl Printer {
                     )?;
                 }
 
-                if any_fixable {
-                    writeln!(writer, "[*] fixable with `ruff check --fix`",)?;
-                }
+                self.write_summary_text(writer, diagnostics)?;
                 return Ok(());
             }
             OutputFormat::Json => {

--- a/crates/ruff/tests/integration_test.rs
+++ b/crates/ruff/tests/integration_test.rs
@@ -950,15 +950,40 @@ fn rule_invalid_rule_name() {
 #[test]
 fn show_statistics() {
     let mut cmd = RuffCheck::default()
-        .args(["--select", "F401", "--statistics"])
+        .args(["--select", "C416", "--statistics"])
         .build();
     assert_cmd_snapshot!(cmd
-        .pass_stdin("import sys\nimport os\n\nprint(os.getuid())\n"), @r"
+                         .pass_stdin(r#"
+def mvce(keys, values):
+    return {key: value for key, value in zip(keys, values)}
+"#), @r"
     success: false
     exit_code: 1
     ----- stdout -----
-    1	F401	[*] unused-import
-    [*] fixable with `ruff check --fix`
+    1	C416	unnecessary-comprehension
+    Found 1 error.
+    No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
+
+    ----- stderr -----
+    ");
+}
+
+#[test]
+fn show_statistics_unsafe_fixes() {
+    let mut cmd = RuffCheck::default()
+        .args(["--select", "C416", "--statistics", "--unsafe-fixes"])
+        .build();
+    assert_cmd_snapshot!(cmd
+                         .pass_stdin(r#"
+def mvce(keys, values):
+    return {key: value for key, value in zip(keys, values)}
+"#), @r"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    1	C416	[*] unnecessary-comprehension
+    Found 1 error.
+    [*] 1 fixable with the --fix option.
 
     ----- stderr -----
     ");
@@ -969,21 +994,57 @@ fn show_statistics_json() {
     let mut cmd = RuffCheck::default()
         .args([
             "--select",
-            "F401",
+            "C416",
             "--statistics",
             "--output-format",
             "json",
         ])
         .build();
     assert_cmd_snapshot!(cmd
-        .pass_stdin("import sys\nimport os\n\nprint(os.getuid())\n"), @r#"
+        .pass_stdin(r#"
+def mvce(keys, values):
+    return {key: value for key, value in zip(keys, values)}
+"#), @r#"
     success: false
     exit_code: 1
     ----- stdout -----
     [
       {
-        "code": "F401",
-        "name": "unused-import",
+        "code": "C416",
+        "name": "unnecessary-comprehension",
+        "count": 1,
+        "fixable": false
+      }
+    ]
+
+    ----- stderr -----
+    "#);
+}
+
+#[test]
+fn show_statistics_json_unsafe_fixes() {
+    let mut cmd = RuffCheck::default()
+        .args([
+            "--select",
+            "C416",
+            "--statistics",
+            "--unsafe-fixes",
+            "--output-format",
+            "json",
+        ])
+        .build();
+    assert_cmd_snapshot!(cmd
+        .pass_stdin(r#"
+def mvce(keys, values):
+    return {key: value for key, value in zip(keys, values)}
+"#), @r#"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    [
+      {
+        "code": "C416",
+        "name": "unnecessary-comprehension",
         "count": 1,
         "fixable": true
       }

